pkg/controller/backup_controller.go

```diff
@@ -56,23 +62,28 @@ import (
 type backupController struct {
 	*genericController
 
-	backupper                pkgbackup.Backupper
-	lister                   velerov1listers.BackupLister
-	client                   velerov1client.BackupsGetter
-	clock                    clock.Clock
-	backupLogLevel           logrus.Level
-	newPluginManager         func(logrus.FieldLogger) clientmgmt.Manager
-	backupTracker            BackupTracker
-	backupLocationLister     velerov1listers.BackupStorageLocationLister
-	defaultBackupLocation    string
-	defaultBackupTTL         time.Duration
-	snapshotLocationLister   velerov1listers.VolumeSnapshotLocationLister
-	defaultSnapshotLocations map[string]string
-	metrics                  *metrics.ServerMetrics
-	newBackupStore           func(*velerov1api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error)
-	formatFlag               logging.Format
+	backupper                   pkgbackup.Backupper
+	lister                      velerov1listers.BackupLister
+	client                      velerov1client.BackupsGetter
+	csiClient                   snapshotv1beta1.SnapshotV1beta1Interface
+	clock                       clock.Clock
+	backupLogLevel              logrus.Level
+	newPluginManager            func(logrus.FieldLogger) clientmgmt.Manager
+	backupTracker               BackupTracker
+	backupLocationLister        velerov1listers.BackupStorageLocationLister
+	defaultBackupLocation       string
+	defaultBackupTTL            time.Duration
+	snapshotLocationLister      velerov1listers.VolumeSnapshotLocationLister
+	defaultSnapshotLocations    map[string]string
+	metrics                     *metrics.ServerMetrics
+	newBackupStore              func(*velerov1api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error)
+	formatFlag                  logging.Format
+	volumeSnapshotLister        snapshotv1beta1listers.VolumeSnapshotLister
+	volumeSnapshotContentLister snapshotv1beta1listers.VolumeSnapshotContentLister
 }
 
+// TODO(nrb-csi): Add clients for the VS/VSContent here.
+// How about creating them in the server iff EnableCSI, and if they're nil in the backup controller, don't worry about it?
```

Comment:
that seems reasonable to me.

Reply:
I believe with the listers, we don't need the clients now, but I think similar logic applies, and that's what I tried to follow.